pkg/controller/util.go

```diff
@@ -137,13 +137,19 @@ Users:
 					c.logger.Warningf("unknown key %q", p)
 				}
 			}
+
+			delete(data, key)
 		}
 
 		if t.Name != "" {
 			result[t.Name] = t
 		}
 	}
 
+	if len(data) != 0 {
+		c.logger.Warningf("Unprocessed entries in the infrastructure roles' secret. That may mean infrastructure-roles.yaml has incorrect format; see the file in the Postgres operator repo for the expected format.")
```

Comment:
I'd show the number of entries and not mention the manifest file in the comments, since it may have a different name, also we may move the example file and forget to change this message. Something like (you can also avoid hard-coding the list of allowed keys):

```
c.logger.Warningf("%d unprocessed entries for the infrastructure roles", len(data))
c.logger.Info(`infrastructure role entries should be in the {key}{id} format, where {key} can be either of "user", "password", "inrole" and the {id} a monotonically increasing integer starting with 1`)
c.logger.Debugf("unprocessed entries: %#v", data)
```